### ISSUE TITLE
[Doc] update the guide to enforce plugin name and pluginArgs

### ISSUE THREAD

**Issue by Huang-Wei** [kubernetes-sigs/scheduler-plugins#186]
The upstream scheduler decodes the pluginArgs withing the naming conversion: "plugin name" + "Args":

https://github.com/kubernetes/kubernetes/blob/74497632f4118e7989da73de0ed6f1268aeb64ad/pkg/scheduler/framework/runtime/framework.go#L421

So if out-of-tree plugins rely on pluginArgs, its plugin name and name of pluginArgs struct should also be following this naming pattern. Otherwise the decoding logic would throw some error like:

```
    --- FAIL: TestSetup/single_profile_config_-_Network_Traffic_with_args (0.95s)
        main_test.go:599: couldn't create scheduler using provider "DefaultProvider": initializing profiles: creating profile for scheduler name default-scheduler: error initializing plugin "XYZ": want args to be of type runtime.Unkown, got *runtime.Unknown
```

Ref: https://kubernetes.slack.com/archives/C09TP78DV/p1618423441078300?thread_ts=1618378191.077700&cid=C09TP78DV

**Comment by lianghao208** [comment]
I would like to fix this issue.

**Comment by seanmalloy** [comment]
> I would like to fix this issue.

@lianghao208 awesome! Thanks for your help. It is greatly appreciated.

/assign @lianghao208

**Comment by seanmalloy** [comment]
Looks like https://github.com/kubernetes-sigs/scheduler-plugins/pull/187 is attempted to resolve this issue.

**Comment by lianghao208** [comment]
@seanmalloy Yes, I noticed that

**Comment by seanmalloy** [comment]
Closed by #187.

/close
